src/net.md

```diff
@@ -831,6 +832,57 @@ shows Listening on 127.0.0.1:11500, run
 
 After sending data in telnet press `ctrl-]` and type `quit`.
 
+[ex-extract-links-webpage]: #ex-extract-links-webpage
+<a name="ex-extract-links-webpage"/>
+## Extract all links from a webpage
+
+[![reqwest-badge]][reqwest] [![select-badge]][select] [![cat-net-badge]][cat-net]
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+extern crate select;
+
+use std::io::Read;
+use select::document::Document;
+use select::predicate::Name;
+
+# error_chain! {
+#    foreign_links {
+#        ReqError(reqwest::Error);
+#        IoError(std::io::Error);
+#    }
+# }
+
+fn run() -> Result<()> {
+    let mut res = reqwest::get("https://www.rust-lang.org/en-US/")?;
```

Comment:
`Reqwest::Response` implements `io::Read` I would suggest using `Document::from_read` in such way you can drop all the intermediate "buffer" and "html" lines leaving just
```rust
    let res = reqwest::get("https://www.rust-lang.org/en-US/")?;
    let document = Document::from_read(res)?;
```